cmd/keytransparency-delegate/main.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"database/sql"
+	"flag"
+	"log"
+	"net/http"
+
+	"github.com/google/keytransparency/cmd/serverutil"
+	"github.com/google/keytransparency/core/managementserver"
+	"github.com/google/keytransparency/impl/sql/engine"
+	"github.com/google/keytransparency/impl/sql/keysets"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/reflection"
+
+	pb "github.com/google/keytransparency/core/api/usermanager/v1/usermanager_proto"
+	_ "github.com/google/trillian/crypto/keys/der/proto"
+	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
+)
+
+var (
+	addr         = flag.String("addr", ":8080", "The ip:port combination to listen on")
+	metricsAddr  = flag.String("metrics-addr", ":8081", "The ip:port to publish metrics on")
+	serverDBPath = flag.String("db", "test:zaphod@tcp(localhost:3306)/test", "Database connection string")
+	keyFile      = flag.String("tls-key", "genfiles/server.key", "TLS private key file")
+	certFile     = flag.String("tls-cert", "genfiles/server.crt", "TLS cert file")
+
+	instance = flag.Int64("instance", 0, "Instance number. Typically 0.")
+)
+
+func openDB() *sql.DB {
```

Comment:
I think I'd return `(*sql.DB, error)` here and have another `glog.Exitf()` in the `main()` as you do for things that follow the `openDB()` call.

Reply:
Clean code is good code :-)